File: Task2/Task2.rb
def counter(nums)
    count = 0
    nums.each do |num|
        digits = (num == 0) ? 1 : 0
        n = num.abs
        while n != 0
            n /= 10
            digits += 1
        end
        count += 1 if digits.odd?
    end
    count
end

print "Введите числа через запятую: "
input = gets.chomp
nums = input.split(',').map(&:to_i)
puts counter(nums)

File: Task2/Task2.py
def counter(nums):
    count = 0
    for num in nums:
        digits = 1 if num == 0 else 0
        n = abs(num)
        while n != 0:
            n //= 10
            digits += 1
        if digits % 2 != 0:
            count += 1
    return count

input_str = input("Введите числа через запятую: ")
nums = list(map(int, input_str.split(',')))
print(counter(nums))